docs/tutorial/07_setting-up-the-front-end.md

```diff
@@ -0,0 +1,70 @@
+# 7. Setting up the front-end
+
+With the backend contract written and tested, the next step is to have a front end for users to interact with it. Typically this front end is a web page executing
+JavaScript. This way the user can interact with the decentralized application (dapp) and the Ethereum blockchain without having to rely on the web server. For this tutorial 
+we chose to use Facebook's [React Framework](https://reactjs.org/tutorial/tutorial.html).
+
+Browsers don't have native support for Ethereum yet. To use a dapp users need to install a wallet browser extension, such as [MetaMask](https://metamask.io/). 
+This kind of extension injects an Ethereum provider object into the browse that can be accessed from JavaScript code running in a web page.
```

Comment:
> injects an Ethereum provider 

This concept hasn't been explained. I think this can be confusing. Shouldn't we add an explanation, or at least mention that it will be explained later?

Reply:
Next commit this will become:

"Browsers don't have native support for Ethereum yet. To use a dapp users need to install a wallet browser extension, such as [MetaMask](https://metamask.io/). 
Wallet browser extensions provide the JavaScript running inside the web page with an object, called a provider, that is used to access the API."